src/ui/public/courier/__tests__/request_queue.js

```diff
@@ -3,82 +3,37 @@ import expect from 'expect.js';
 import sinon from 'sinon';
```

Comment:
Yea, but they'd probably be as useless as these current tests. :)  All these tests are checking is if canStart is called, since it's mocked out.  Not really actually checking any of the logic in the actual canStart function (actually I now notice you point this out in your comment below).  So likewise, I'd just be manually setting started = true/false on the mock, and would just make sure the filter returns the right array (if I followed suit).

My concern with adding better tests at this point is that I'd be testing that my current implementation works, not that my implementation matches the old implementation, and if I make any wrong assumptions it could be confusing. 

For instance, a better test I'd think of adding would be something like `getStartable doesn't return already started requests`.  But as I dig deeper, `canStart` doesn't even check the `this.started` variable.  Is that a bug in the old implementation?  The function `start` doesn't even set `_fetchRequested` to true, that's a separate call.  So the class and function flow doesn't make much sense to me. To write tests to assert it behaves this way, I'm afraid will be misleading.  Similar to how you saw the test that asserted $ is removed from the JSON object.  As soon as you see a test for that, you think, well this test was added for a reason so this must be important, lets make sure going forward, it always stays this way.  But maybe we don't even need to do that anymore. 

Maybe I should just get rid of this test file since it's really only testing the mock, like you mentioned below.  What do you think?

Reply:
I think if we're going to have a test for `.getStartable()` then we should also have a test for `getInactive()`. Why this wasn't done previously, I'm not  exactly sure; however, the old implementation utilized `.get()` which was "under test", so this gave us some sense of certainty. 

> Maybe I should just get rid of this test file since it's really only testing the mock, like you mentioned below. What do you think?

I'm not seeing how the `getStartable()` or the potential `getInactive()` test are testing the behavior of the mock. The previous test, that was deleted, was testing the default behavior of the mock.

With all that being said, I don't feel strongly about us having tests for this "class" in general, as it does little to guarantee the correctness of the courier in general.

Reply:
added with a warning message :)